### PR TITLE
Made tracing of proxy symints lazy

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85185
* #84918

### CODE REVIEW

File: torch/fx/experimental/proxy_tensor.py [hunk 1]
```diff
@@ -104,25 +104,34 @@ def set_meta(proxy, val):
             proxy.node.meta['tensor_meta'] = _extract_tensor_metadata(val)
     return proxy
 
+def thunkify(f, *args, **kwargs):
+    """
+    Delays computation of f until it's called again, which is done by
```
Comment (ezyang) [inline torch/fx/experimental/proxy_tensor.py:109]: Done by the what

File: torch/fx/experimental/proxy_tensor.py [hunk 2]
```diff
@@ -104,25 +104,34 @@ def set_meta(proxy, val):
...
+    """
+    return functools.lru_cache(1)(functools.partial(f, *args, **kwargs))
```
Comment (ezyang) [inline torch/fx/experimental/proxy_tensor.py:111]: lru_cache boooooo (can be faster if done manually)

File: torch/fx/experimental/proxy_tensor.py [hunk 3]
```diff
@@ -104,25 +104,34 @@ def set_meta(proxy, val):
...
 
 def track_tensor(tensor, proxy, *, constant, tracer):
+    def try_set_proxy_slot(outer_s, proxy_callable, *args):
+        assert callable(proxy_callable)
+        if isinstance(outer_s, SymInt):
+            inner_s = outer_s.get_pyobj()
+            assert isinstance(inner_s, PySymInt)
+            proxy = None
```
Comment (ezyang) [inline torch/fx/experimental/proxy_tensor.py:119]: This is dead

**Review by ezyang — APPROVED** [review]

File: torch/fx/experimental/proxy_tensor.py (hunk 2 above)
Comment (Chillee) [reply torch/fx/experimental/proxy_tensor.py:111]: If you remember, I used to manually implement it. But I couldn't benchmark a difference in my testing lol.

**Comment by ezyang** [comment]
There's a decent chance it's infinite looping in tests

**Comment by Chillee** [comment]
@ezyang of a sorts lol

image

**Pull request closed without merging**